src/helium_follower_service.erl

```diff
@@ -54,8 +55,50 @@ subnetwork_last_reward_height(Ctx, Req) ->
 
 -spec active_gateways(follower_pb:follower_gateway_stream_req_v1_pb(), grpcbox_stream:t()) ->
         {ok, grpcbox_stream:t()} | grpcbox_stream:grpc_error_response().
+active_gateways(#follower_gateway_stream_req_v1_pb{} = _Msg, StreamState) ->
+    #{chain := Chain} =  grpcbox_stream:stream_handler_state(StreamState),
+    case Chain of
+        undefined ->
+            lager:debug("chain not ready, returning error response for msg ~p", [_Msg]),
+            {grpc_error, {grpcbox_stream:code_to_status(14), <<"temporarily unavailable">>}};
+        _ ->
+            grpcbox_stream:update_headers([{<<"gateway_stream">>, <<"start">>}], StreamState),
+
+            Ledger = blockchain:ledger(Chain),
+            {ok, Height} = blockchain_ledger_v1:current_height(Ledger),
+
+            InteractiveBlock = case ?get_var(?hip17_interactivity_blocks, Ledger) of
+                                   {ok, V} -> V;
+                                   {error, not_found} -> 0
+                               end,
+            lager:info("LOADING ACTIVE GATEWAYS", []),
+            NumGateways =
+                blockchain_ledger_v1:cf_fold(
+                    active_gateways,
+                    fun({Addr, BinGw}, Acc) ->
+                        Gw = blockchain_ledger_gateway_v2:deserialize(BinGw),
+                        Loc = blockchain_ledger_gateway_v2:location(Gw),
+                        LastChallenge = blockchain_ledger_gateway_v2:last_poc_challenge(Gw),
+                        case Loc /= undefined andalso (LastChallenge /= undefined andalso (Height - LastChallenge) =< InteractiveBlock) of
+                            true ->
+                                Msg = #follower_gateway_resp_v1_pb{
+                                          height = Height,
+                                          address = Addr,
+                                          location = h3:to_string(Loc),
+                                          owner = blockchain_ledger_gateway_v2:owner_address(Gw)
+                                      },
+                                grpcbox_stream:send(false, Msg, StreamState),
```

Comment:
'send' also returns an updated state and as you are sending multiple msgs here, will need to handle that each iteration